hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -329,6 +316,23 @@ private void saveInternalSchema(HoodieTable table, String instantTime, HoodieCom
     }
   }
 
+  void runTableServicesInline(Option<Map<String, String>> extraMetadata, HoodieTable table, HoodieCommitMetadata metadata) {
+    // We don't want to fail the commit if hoodie.fail.writes.on.inline.table.service.exception is false. We catch warn if false
+    try {
+      // trigger clean and archival.
+      // Each internal call should ensure to lock if required.
+      mayBeCleanAndArchive(table);
+      // do this outside of lock since compaction, clustering can be time taking and we don't need a lock for the entire execution period
+      runCompactionClusteringInline(table, metadata, extraMetadata);
+    } catch (Exception e) {
+      if (config.isFailOnInlineTableServiceExceptionEnabled()) {
+        throw e;
+      }
+      LOG.warn("Inline table services failed with exception: " + e.getMessage()
```

Comment:
This log will lose all the context of the exception. When logging a warning, you can also include the exception as the last arg so you can view the stacktrace.